controllers/restic.go

```diff
@@ -82,40 +82,39 @@ func (r *VolumeSnapshotBackupReconciler) CreateVSBResticSecret(log logr.Logger)
 	var scheduleCronExpr = ""
 	rpolicy := RetainPolicy{}
 	for key, val := range resticSecret.Data {
+		stringVal := string(val)
 		if key == ResticRepository {
 			// if trailing '/' in user-created Restic repo, remove it
-			stringVal := string(val)
-			stringVal = strings.TrimRight(stringVal, "/")
-
-			ResticRepoValue = stringVal
+			ResticRepoValue = strings.TrimRight(stringVal, "/")
 		}
 		if key == ResticPruneInterval {
-			pruneInterval = string(val)
+			pruneInterval = stringVal
 		}
 		if key == SnapshotScheduleCron {
-			scheduleCronExpr = string(val)
+			scheduleCronExpr = stringVal
 		}
 		if key == SnapshotRetainPolicyMonthly {
-			rpolicy.monthly = string(val)
+			rpolicy.monthly = stringVal
 		}
 		if key == SnapshotRetainPolicyDaily {
-			rpolicy.daily = string(val)
+			rpolicy.daily = stringVal
 		}
 		if key == SnapshotRetainPolicyHourly {
-			rpolicy.hourly = string(val)
+			rpolicy.hourly = stringVal
 		}
 		if key == SnapshotRetainPolicyWeekly {
-			rpolicy.weekly = string(val)
+			rpolicy.weekly = stringVal
 		}
 		if key == SnapshotRetainPolicyYearly {
-			rpolicy.yearly = string(val)
+			rpolicy.yearly = stringVal
 		}
 		if key == SnapshotRetainPolicyWithin {
-			rpolicy.within = string(val)
+			rpolicy.within = stringVal
 		}
 	}
 
-	resticrepo := fmt.Sprintf("%s/%s/%s", ResticRepoValue, pvc.Namespace, pvc.Name)
+	// TODO should check if label is present first? or it is always present?
```

Comment:
Do you know if this label is always present?

Reply:
I tested with normal backup, backup from schedule and also did a restore (this one doesn't matter, right?), and did not had problems. Do you remember another scenarios?

Reply:
If the label is not found, it will be an empty string, I assume though that we are always setting this, so the only case is if a user manually creates the resource. @shubham-pampattiwar can you make sure that this understanding is correct.

as long as the repo can handle "/resticname/namespace//name" it should be fine

Reply:
I will test what happens putting a trailing slash `/` in the name

One thing that I thought is putting a validator in the yaml for the backup name, like minLength and pattern https://swagger.io/docs/specification/data-models/data-types/#string, then we say for example that it has to be at least of length 1 and can't have slashes in it (but I do not where is this yaml)

Reply:
@shawn-hurley @mateusoliveira43 we set the backup label during the creation of VSB CR via velero-plugin-for-vsm: https://github.com/migtools/velero-plugin-for-vsm/blob/main/internal/backup/volumesnapshotcontent_action.go#L118

Reply:
from the CLI I was not able to trigger an error (I do not know how to use yamls to do the same)

```sh
$ velero backup create "beckup/" --include-namespaces mysql-persistent
An error occurred: Backup.velero.io "beckup/" is invalid: metadata.name: Invalid value: "beckup/": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
command terminated with exit code 1
$ velero backup create " " --include-namespaces mysql-persistent
An error occurred: Backup.velero.io " " is invalid: metadata.name: Invalid value: " ": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Reply:
@mateusoliveira43 you [can follow here](https://velero.io/docs/v1.11/api-types/backup/) to see how to form the backup yaml.

Reply:
So tried one last thing, create VSB by hand without the label, but still no bug (to trigger a possible bug, we would have to create the VSB at a specific moment and with the snapcontent name, so I think is safe)

```yaml
apiVersion: datamover.oadp.openshift.io/v1alpha1
kind: VolumeSnapshotBackup
metadata:
  generateName: vsb-
  generation: 1
  labels:
    datamover.io/vsb-volumesnapshotcontent: snapcontent-8281b69d-f406-486e-8a68-40996de22e4e
    # velero.io/backup-name: backup-name
  namespace: mysql-persistent
spec:
  protectedNamespace: openshift-adp
  resticSecretRef:
    name: oadp-vsm-example-1-volsync-restic
  volumeSnapshotContent:
    name: snapcontent-8281b69d-f406-486e-8a68-40996de22e4e
```
going to remove TODO to close the PR :)